### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by LeanderK**
- the database now checks whether the worker is allowed to submit ratings/answers
- some fixes in the rest-package

### CODE REVIEW

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java [hunk 1]
```diff
@@ -37,6 +40,14 @@ public AnswerRatingOperations(DSLContext create, CalibrationOperations calibrati
      */
     public AnswerRecord insertNewAnswer(AnswerRecord answerRecord) {
         answerRecord.setIdAnswer(null);
+        ExperimentRecord experiment = experimentOperations.getExperiment(answerRecord.getExperiment())
+                .orElseThrow(() -> new IllegalArgumentException("Illegal experiment-value in rating record."));
+        if (getAnswerCount(answerRecord.getWorkerId()) >= experiment.getAnwersPerWorker()) {
+            throw new IllegalStateException(
```
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java:48]: Should be mapped to a `BadRequestException`.

Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java:48]: but not in database

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/ExperimentResource.java [hunk 2]
```diff
@@ -334,8 +334,7 @@ private Experiment updateExperimentInfo(int id, Experiment experiment, Experimen
 
         experiment.getAlgorithmQualityRating().getParametersList().forEach(param -> algorithmsOperations.storeRatingQualityParam(id, param.getId(), param.getValue()));
 
-        //TODO fix!
-        if (!Objects.equals(old.getTemplateId() , experimentRecord.getTemplate())) {
+        if (experiment.hasTemplateId() &&  (!old.hasTemplateId() || !Objects.equals(old.getTemplateId().getValue() , experimentRecord.getTemplate()))) {
```
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/ExperimentResource.java]: Why does the template ID still delete the rating options?

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java (hunk 1 above)
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java:48]: If you add an exception there, please update the rest call to catch it.

Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java:48]: yeah i forgot that, it's also missing in the doc

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java [hunk 3]
```diff
@@ -22,21 +23,33 @@
...
      * @param answerRecord the record to insert
      * @return the resulting record
+     * @throws IllegalArgumentException if the experiment the answers is referring to is not existing
+     * @throws IllegalStateException if the worker is not allowed to submit more answers
      */
-    public AnswerRecord insertNewAnswer(AnswerRecord answerRecord) {
+    public AnswerRecord insertNewAnswer(AnswerRecord answerRecord) throws IllegalArgumentException, IllegalStateException {
         answerRecord.setIdAnswer(null);
+        ExperimentRecord experiment = experimentOperations.getExperiment(answerRecord.getExperiment())
+                .orElseThrow(() -> new IllegalArgumentException("Illegal experiment-value in rating record."));
```
Comment (simonkorz) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java]: Error message might be misleading. Shouldn't it say "Illegal experiment-id in answer record."

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java [hunk 4]
```diff
@@ -22,21 +23,33 @@
...
+        if (getAnswerCount(answerRecord.getWorkerId()) >= experiment.getAnwersPerWorker()) {
+            throw new IllegalStateException(
+                    String.format("Worker %d already submitted the maximum of allowed ratings", answerRecord.getWorkerId())
```
Comment (simonkorz) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java]: copy paste? Shouldn't it be "allowed answers"?

**Pull request merged**